modules/post/linux/gather/checkcontainer.rb

```diff
@@ -33,6 +33,32 @@ def run
       container = 'Docker'
     end
 
+    # Check for .dockerinit file
+    if container.nil? && file?('/.dockerinit')
+      container = 'Docker'
+    end
+
+    # Check for /.containerenv file
+    if container.nil? && file?('/.containerenv')
+      container = 'Podman'
+    end
+
+    # Check for /dev/lxd/sock file
+    if container.nil? && file?('/dev/lxd/sock')
```

Comment:
```suggestion
    if container.nil? && directory?('/dev/lxc')
```
I'm not seeing a file `sock` inside `/dev/lxc/`

<details>
<summary>Testing in metasploit</summary>

<p>

```

msf6 post(linux/gather/checkcontainer) > rexploit
[*] Reloading module...

From: /home/msfuser/git/metasploit-framework/modules/post/linux/gather/checkcontainer.rb:51 Msf::Modules::Post__Linux__Gather__Checkcontainer::MetasploitModule#run:

    46:
    47:     require 'pry-byebug'
    48:     binding.pry
    49:
    50:     # Check for /dev/lxd/sock file
 => 51:     if container.nil? && file?('/dev/lxd/sock')
    52:       container = 'LXC'
    53:     end
    54:
    55:     # Check for WSL, as suggested in https://github.com/Microsoft/WSL/issues/423#issuecomment-221627364
    56:     if container.nil? && file?('/proc/sys/kernel/osrelease')

[1] pry(#<Msf::Modules::Post__Linux__Gather__Checkcontainer::MetasploitModule>)> file?('/dev/lxd/sock')
=> false
[10] pry(#<Msf::Modules::Post__Linux__Gather__Checkcontainer::MetasploitModule>)> directory?('/dev/lxc')
=> true

```

</p>

</details>

<details>

<summary>Contents of /dev/lxc</summary>

<p>

```
root@ubuntu-c1:/dev/lxc# ls -l
total 0
crw--w---- 1 root tty 136, 0 Jul 18 22:38 console
crw--w---- 1 root tty 136, 1 Jul 18 22:38 tty1
crw--w---- 1 root tty 136, 2 Jul 18 22:38 tty2
crw--w---- 1 root tty 136, 3 Jul 18 22:38 tty3
crw--w---- 1 root tty 136, 4 Jul 18 22:38 tty4
```
</p>

</details>


<details>

<summary>Container setup</summary>

<p> 

```
sudo lxc-create -t download -n ubuntu-c1 -- -d ubuntu -r xenial -a amd64
sudo lxc-start -n ubuntu-c1
sudo lxc-attach -n ubuntu-c1

root@ubuntu-c1:/# apt-get install openssh-server -y
root@ubuntu-c1:/# echo 'PermitRootLogin: yes' >> /etc/ssh/sshd_config
root@ubuntu-c1:/# service sshd restart
root@ubuntu-c1:/# echo 'root:notpassword' | chpasswd
```

</p>

</details>